components/org.wso2.carbon.identity.oauth.uma.grant/src/main/java/org/wso2/carbon/identity/oauth/uma/grant/UMA2GrantHandler.java

```diff
@@ -122,6 +122,7 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws Id
         if (isValidTicket) {
             AuthenticatedUser authenticatedUser = new AuthenticatedUser();
             authenticatedUser.setUserName(subject);
+            authenticatedUser.setTenantDomain(applicationTenantDomain);
```

Comment:
What abt the user store domain?
Shouldn't we set that as well?

Reply:
userstore domain is not expected in the flow. So hope that is not necessary to be set

Reply:
When we create the user object it would be better to set the user store domain. Otherwise, any code that consumes the user object would assume the user is from PRIMARY userstore domain if not set.

Reply:
The problem is by default the subject won't contain the user store domain appended. We should enforce the property 'Use user store domain in local subject identifier' to be enabled to get the user store domain from the subject Id.
BTW I have done the change to extract user store domain from the subject and set to the user object. If the subject doesn't contain the user store domain, then the user store domain will be set as PRIMARY. But we can enforce in the doc to enable the property to set user store domain. We have decided to do the same for tenant domain aswell.